x-pack/plugins/ml/public/application/components/ml_inference/components/test_pipeline.tsx

```diff
@@ -202,12 +329,24 @@ export const TestPipeline: FC<Props> = memo(({ state, sourceIndex }) => {
               <EuiFlexItem grow={false}>
                 <EuiButtonEmpty
                   size="xs"
-                  onClick={getSampleDocs}
+                  onClick={() => setSampleDocsString(lastFetchedSampleDocsString)}
                   disabled={sampleDocsString === ''}
                 >
                   {i18n.translate(
                     'xpack.ml.trainedModels.content.indices.pipelines.addInferencePipelineModal.steps.test.resetSampleDocsButton',
-                    { defaultMessage: 'Reset sample docs' }
+                    { defaultMessage: 'Reset' }
+                  )}
+                </EuiButtonEmpty>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButtonEmpty
+                  size="xs"
+                  onClick={getRandomSampleDoc}
```

Comment:
I think the Result pane should be cleared when the sample doc is reloaded.

Reply:
Updated in 2c51898d8c7022455d87e5815e92eb7880e092f9